code/game/gamemodes/marker/gamemodes.dm

```diff
@@ -123,4 +117,20 @@ GLOBAL_DATUM_INIT(shipsystem, /datum/ship_subsystems, new)
 	if (GM.marker_active)
 		to_chat(src, "The marker is already active")
 		return
-	GM.activate_marker()
\ No newline at end of file
+	GM.activate_marker()
+
+/**
+
+There is probably a better way to do this, but I've added some more stringent checks to the gamemode win conditions to prevent things like:
+Admin characters being counted as crew, and on their deletion, ending the game
+Non-critical characters like any ghost-roles you may wish to add, or even antags, from counting as "dead crew" (Antags fallback to their own "are the antags dead checks"
+
+*/
+
+//Method to be called when a human player dies for any reason, this will check the amount of players that should be alive, and sees if they're dead or not. We're doing it via an almost signal (bay lacks signals) to avoid processing overhead.
+/datum/game_mode/marker/check_finished()
```

Comment:
i dont see the base class defininition of this function, did it already exist?

Reply:
It's a standard gamemode proc called on tick to check if the mode should end or not, when check finished returns TRUE, the round ends